### ISSUE TITLE
Support non rectangular rooms

### ISSUE THREAD

**Issue by jwvanderbeck** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#311]
### Description

Digging through the code, it looks like a Room is just essentially a named Zone which means it is always rectangular.  The actual vacuum and accompanying app however support non rectangular rooms.  It seems to me a Room should actually be a list of Zones?

### Solution

I would like to be able to extract the rooms as defined in the original app which are not neccesarily rectangular.

### Alternatives

I THINK I can get this working at the user level by manually defining multiple zones as one cleaning task but it's kind of dirty (pun not intended) and doesn't let me actually put the "room" into the Rooms list on the companion Lovelace card which is confusing.  I'm also not 100% sure I can even get it working.  Still messing with this possible workaround.

### Context

![image](https://user-images.githubusercontent.com/161126/174813880-dd4ad003-0926-4c01-8f17-39b4ffe58a73.png)
![image](https://user-images.githubusercontent.com/161126/174813939-b2994216-3a70-4487-8921-664143088def.png)

**Comment by jwvanderbeck** [comment]
This is exactly the process I did and that is what the second screenshot above shows.  Which do not match the actual rooms as defined on the app. You will see the "extracted" rooms are just rectangles that don't actually match the rooms.  Notice the yellow room 16 for example.  That is an n-gon but the extracted room is just one big rectangle around the whole thing.

**Comment by jwvanderbeck** [comment]
I can move this to your card if that is better but I think the underlying issue is in the EXTRACTOR as it defines a room as a rectangle, so there isn't really anything more you could do in the card unless I am missing something.

**Comment by PiotrMachowski** [comment]
No, rooms are defined by ID, outline is just for presentation purposes. You can delete it or modify to your liking (check out demo config).

**Comment by jwvanderbeck** [comment]
Ahh so when my vacuum is told to clean, it is just told "clean room 16".
Still a shame the extractor can't get the proper room outline from the map allowing me to display it in the card.  I'll go dig around in both of these some more and see what I can figure out.

**Comment by PiotrMachowski** [comment]
Yup, just room number is passed to the vacuum. To get proper room outline map extractor would need to perform some kind of image analysis and I don't really feel like implementing it. You can manually create an outline in any shape to adjust to your rooms layout.
